src/views/AddLocusGeneDiseaseView.vue

```diff
@@ -144,41 +148,102 @@ export default {
+        return;
+      }
+      // if inputPmids is not empty then continue further
+      this.isInputPmidsValid = true;
+      this.publicationsErrorMsg = null;
       this.isPublicationsDataLoading = true;
-      let pmidListStr = inputPmids
+      let pmidListStr = this.inputPmids
+        .trim()
         .split(";")
         .filter((item) => item)
         .join(",");
+      let responseStatus = null;
       fetch(`/gene2phenotype/api/publication/${pmidListStr}/`)
         .then((response) => {
-          if (response.status === 200) {
-            return response.json();
-          } else {
-            return Promise.reject(
-              new Error("Unable to fetch publications data")
-            );
-          }
+          responseStatus = response.status;
+          return response.json();
         })
         .then((responseJson) => {
           this.isPublicationsDataLoading = false;
-          this.publicationsData = responseJson;
-          if (this.publicationsData && this.publicationsData.results) {
-            this.input = updateInputWithPublicationsData(
-              this.input,
-              this.publicationsData
-            );
+          if (responseStatus === 200) {
+            const publicationsData = responseJson;
+            if (publicationsData && publicationsData.results) {
+              this.input = updateInputWithPublicationsData(
+                this.input,
+                publicationsData
+              );
+              let pmidList = publicationsData.results.map((item) => item.pmid);
+              this.hpoTermsInputHelper =
+                updateHpoTermsInputHelperWithPublicationsData(pmidList);
+            }
+          } else if (responseStatus === 404) {
+            this.publicationsErrorMsg = responseJson.detail
+              ? responseJson.detail
+              : "Unable to fetch publications data.";
+            console.log(this.publicationsErrorMsg);
+          } else {
+            this.publicationsErrorMsg = "Unable to fetch publications data.";
+            console.log(this.publicationsErrorMsg);
           }
         })
         .catch((error) => {
           this.isPublicationsDataLoading = false;
-          this.publicationsErrorMsg = error.message;
+          this.publicationsErrorMsg = "Unable to fetch publications data.";
+          console.log(error);
+        });
+    },
+    fetchHpoTerms(pmid) {
+      // if hpoTermsInput is empty then set isHpoTermsValid to false and dont continue further
+      if (this.hpoTermsInputHelper[pmid].hpoTermsInput.trim() === "") {
+        this.hpoTermsInputHelper[pmid].isHpoTermsValid = false;
+        return;
+      }
+      // if hpoTermsInput is not empty then continue further
+      this.hpoTermsInputHelper[pmid].isHpoTermsValid = true;
+      this.hpoTermsInputHelper[pmid].hpoTermsErrorMsg = "";
+      this.hpoTermsInputHelper[pmid].isHpoTermsDataLoading = true;
+      let hpoTermsListStr = this.hpoTermsInputHelper[pmid].hpoTermsInput
```

Comment:
There is a bug here, the split should also make sure there is no space between semi colons and the next HP0 term. Thank you

Reply:
taken care, the new change will trim any space around the hpo term and between semi colons